components/editor/NoteHeader.tsx

```diff
@@ -39,34 +39,52 @@ export default function NoteHeader() {
     (state) => !state.isSidebarOpen && state.openNoteIds?.[0] === currentNote.id
   );
   const isCloseButtonVisible = useStore(
-    (state) => state.openNoteIds?.[0] !== currentNote.id
+    (state) => state.openNoteIds?.[1]
   );
   const note = useStore((state) => state.notes[currentNote.id]);
 
   const onClosePane = useCallback(() => {
     const currentNoteIndex = store
       .getState()
       .openNoteIds.findIndex((openNoteId) => openNoteId === currentNote.id);
+    const stackedNoteIds = queryParamToArray(stackQuery);
 
     if (currentNoteIndex < 0) {
       return;
     }
 
-    // Remove from stacked notes and shallowly route
-    const stackedNoteIds = queryParamToArray(stackQuery);
-    stackedNoteIds.splice(
-      currentNoteIndex - 1, // Stacked notes don't include the main note
-      1
-    );
-
-    router.push(
-      {
-        pathname: router.pathname,
-        query: { ...router.query, stack: stackedNoteIds },
-      },
-      undefined,
-      { shallow: true }
-    );
+    if (currentNoteIndex == 0) {
+      // Changes Current Note to first note in stack 
+      currentNote.id = stackedNoteIds[0];
+      stackedNoteIds.splice(
+        0, 
+        1
+      );// Removes first note in stack from the stack 
+      router.push(
+        {
+          pathname: router.pathname,
+          query: { ...currentNote, stack: stackedNoteIds },
+        },
+        undefined,
+        { shallow: true }
+      );
+    }
+
+    else {// Remove from stacked notes and shallowly route
+      stackedNoteIds.splice(
+        currentNoteIndex - 1, // Stacked notes don't include the main note
+        1
+      );
+      router.push(
```

Comment:
This is repeated code, so you can probably just pull it out and have it at the end

Reply:
Oh, I misread it. It's actually not repeated - you can leave it!